### PR TITLE
BUG: Handling columns from index_col in _is_potential_multi_index

### LINKED ISSUE #33476: BUG: read_excel fails on multi-index with: empty name or equal names
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

#### Code Sample, a copy-pastable example

```python
import pandas as pd

pd.read_excel('df.xlsx', sheet_name='Sheet1', index_col=0, header=[0, 1])
```

Note: `engine`s could be different.

#### Problem description

Exception on `pd.read_excel()`:
```
$ ~/miniconda3/bin/python read.py 
Traceback (most recent call last):
  File "read.py", line 4, in <module>
    df = pd.read_excel('df_empty.xlsx', engine='openpyxl', sheet_name='Sheet1', index_col=0, header=[0, 1])
  File "/home/sasha/miniconda3/lib/python3.7/site-packages/pandas/io/excel/_base.py", line 334, in read_excel
    **kwds,
  File "/home/sasha/miniconda3/lib/python3.7/site-packages/pandas/io/excel/_base.py", line 888, in parse
    **kwds,
  File "/home/sasha/miniconda3/lib/python3.7/site-packages/pandas/io/excel/_base.py", line 521, in parse
    ].columns.set_names(header_names)
  File "/home/sasha/miniconda3/lib/python3.7/site-packages/pandas/core/indexes/base.py", line 1325, in set_names
    idx._set_names(names, level=level)
  File "/home/sasha/miniconda3/lib/python3.7/site-packages/pandas/core/indexes/base.py", line 1239, in _set_names
    raise ValueError(f"Length of new names must be 1, got {len(values)}")
ValueError: Length of new names must be 1, got 2
```

on XLSX like:
image
(C2 is empty, [df_empty.xlsx](https://github.com/pandas-dev/pandas/files/4464392/df_empty.xlsx))

or this:

image
(B2 == C2, [df_equals.xlsx](https://github.com/pandas-dev/pandas/files/4464393/df_equals.xlsx))

#### Expected Output

No exceptions.

#### Output of ``pd.show_versions()``

<details>
INSTALLED VERSIONS
------------------
commit           : None
python           : 3.7.7.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.4.25-calculate
machine          : x86_64
processor        : Intel(R) Core(TM) i5-7200U CPU @ 2.50GHz
byteorder        : little
LC_ALL           : None
LANG             : ru_RU.utf8
LOCALE           : ru_RU.UTF-8

pandas           : 1.0.3
numpy            : 1.18.1
pytz             : 2019.3
dateutil         : 2.8.1
pip              : 20.0.2
setuptools       : 46.1.3.post20200330
Cython           : 0.29.15
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : 1.2.8
lxml.etree       : 4.5.0
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.11.1
IPython          : 7.13.0
pandas_datareader: None
bs4              : None
bottleneck       : None
fastparquet      : None
gcsfs            : None
lxml.etree       : 4.5.0
matplotlib       : 3.1.3
numexpr          : None
odfpy            : None
openpyxl         : 3.0.3
pandas_gbq       : None
pyarrow          : None
pytables         : None
pytest           : None
pyxlsb           : None
s3fs             : None
scipy            : 1.4.1
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
xlsxwriter       : 1.2.8
numba            : None
</details>

### PR DESCRIPTION
**Pull request by mproszewska**
- [x] closes #33476
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [] whatsnew entry

### CODE REVIEW

File: pandas/io/parsers.py [hunk 1]
```diff
@@ -1168,7 +1168,7 @@ def _is_index_col(col):
     return col is not None and col is not False
 
 
-def _is_potential_multi_index(columns):
+def _is_potential_multi_index(columns, index_col=None):
```
Comment (jreback) [inline pandas/io/parsers.py:1171]: when adding args, can you type them (also existing args if you can)

**Review by jreback — CHANGES_REQUESTED** [review]
can you also add a csv test that hits this

File: pandas/tests/io/parser/test_index_col.py [hunk 2]
```diff
@@ -184,3 +184,19 @@ def test_no_multi_index_level_names_empty(all_parsers):
...
+I12,B,B
+I2,1,3
+"""
+    midx = MultiIndex.from_tuples([("A", "B"), ("A", "B.1")], names=["I11", "I12"])
+    idx = Index(["I2"])
+    expected = DataFrame([[1, 3]], index=idx, columns=midx)
+    result = parser.read_csv(StringIO(data), index_col=0, header=[0, 1])
+    tm.assert_frame_equal(result, expected)
+
+    result = parser.read_csv(StringIO(data), index_col="I11", header=0)
```
Comment (jreback) [inline pandas/tests/io/parser/test_index_col.py:202]: can you compare this last as well

File: pandas/tests/io/excel/test_readers.py [hunk 3]
```diff
@@ -1084,3 +1084,7 @@ def test_excel_high_surrogate(self, engine):
         # should not produce a segmentation violation
         actual = pd.read_excel("high_surrogate.xlsx")
         tm.assert_frame_equal(expected, actual)
+
+    @pytest.mark.parametrize("filename", ["df_empty.xlsx", "df_equals.xlsx"])
+    def test_header_with_index_col(self, engine, filename):
```
Comment (jreback) [inline pandas/tests/io/excel/test_readers.py:1089]: what exactly is this testing? can you add the issue number as a comment. should we also assert the results?

File: pandas/tests/io/parser/test_index_col.py [hunk 4]
```diff
@@ -184,3 +184,25 @@ def test_no_multi_index_level_names_empty(all_parsers):
         expected.to_csv(path)
         result = parser.read_csv(path, index_col=[0, 1, 2])
     tm.assert_frame_equal(result, expected)
+
+
+def test_header_with_index_col(all_parsers):
+    parser = all_parsers
```
Comment (jreback) [inline pandas/tests/io/parser/test_index_col.py:190]: can you add the issue number as a comment

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by mproszewska** [comment]
I added requested changes.

File: pandas/tests/io/excel/test_readers.py (hunk 3 above)
Comment (jreback) [reply pandas/tests/io/excel/test_readers.py:1089]: can you assert the results. as a point of interaction, do not resolve conversions, let the questioner do it.

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by mproszewska** [comment]
Everything should be fine now

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @mproszewska

**Comment by kuraga** [comment]
@mproszewska , thanks very much!

But @mproszewska , @jreback , should we add an argument to [this line](https://github.com/pandas-dev/pandas/pull/33982/files#diff-777d7549579ddc0c6e67596ad87e0d27L1594), too, shouldn't we?

**Comment by kuraga** [comment]
@mproszewska , @jreback , ping.

**Comment by kuraga** [comment]
> should we add an argument to [this line](https://github.com/pandas-dev/pandas/blob/c71bfc36211b5e2d860a06d8fbef902b757bd6e4/pandas/io/parsers.py#L1601), too, shouldn't we?

We shouldn't: we're constructing (multi-)index ourselves there, so we don't have the index column(s)).

**Pull request merged**